mesonbuild/build.py

```diff
@@ -2411,18 +2410,18 @@ def get_target_name(self):
 class ConfigurationData:
     def __init__(self):
```

Comment:
```suggestion
    def __init__(self) -> None:
```

Otherwise it's not type-checked.

Reply:
I just went and read up on this. The rules are absolutely stupid, for anyone else wondering:

If your `__init__` has arguments, and those are typed you may omit the return type and still get typing, if the `__init__` does not  have arguments (other than self), you must include the `-> None`.

Reply:
Wow... In that case, should we just keep it consistent and add `-> None` to every init?

Reply:
> I just went and read up on this. The rules are absolutely stupid, for anyone else wondering

I wouldn't call them stupid, they do make sense.

> Wow... In that case, should we just keep it consistent and add -> None to every init?

This is what I would do. No reason to treat `__init__` as a special case.